R/fetch-bbs-data.R

```diff
@@ -329,13 +331,20 @@ get_birds <- function(level, quiet, connection, force) {
 
   unz_path <- utils::unzip(zipfile = full_path, exdir = tempdir())
 
+  if(release == 2023){ # not zipped in 2023
+    birds <- unz_path %>%
+    purrr::map(readr::read_csv, col_types = "nnnnnnnnnnnnnnn",
+               progress = FALSE) %>%
+    purrr::map(~dplyr::rename_with(.x, snakecase::to_snake_case)) %>%
+    dplyr::bind_rows()
+  }else{
   birds <- unz_path %>%
     purrr::map(utils::unzip, exdir = tempdir()) %>%
     purrr::map(readr::read_csv, col_types = "nnnnnnnnnnnnnnn",
                progress = FALSE) %>%
     purrr::map(~dplyr::rename_with(.x, snakecase::to_snake_case)) %>%
     dplyr::bind_rows()
-
+}
```

Comment:
To reduce redundancy could consider the following (as it looks like all the steps are the same except the unzip).

This can make it easier to to read how things are different.

```
if(release != 2023){ # not zipped in 2023
  unz_path <- purrr::map(unz_path, utils::unzip, exdir = tempdir())
} 

birds <- unz_path %>%
  purrr::map(readr::read_csv, col_types = "nnnnnnnnnnnnnnn",
             progress = FALSE) %>%
  purrr::map(~dplyr::rename_with(.x, snakecase::to_snake_case)) %>%
  dplyr::bind_rows()

```